AdyenNetworking/APIClient/Request.swift

```diff
@@ -55,15 +54,23 @@ public protocol Request: Encodable {
     /// :nodoc:
     /// The HTTP method.
     var method: HTTPMethod { get }
-    
 }
 
-/// Represents an API response.
+/// Describes a ``Request`` extension to be used for async downloading.
+///
+/// A ``DownloadProgressDelegate`` is provided for progress updates.
+@available(iOS 15.0.0, *)
+public protocol AsyncDownloadRequest: Request {
```

Comment:
can just be `DownloadRequest`, it can be reused for non async protocol

Reply:
As mentioned [here](https://developer.apple.com/documentation/foundation/url_loading_system/downloading_files_from_websites) in Apple's docs, the completion handler function does not call delegate methods for download progress. The need for the `AsyncDownloadRequest` is due to the fact that it seems ([here](https://stackoverflow.com/a/70396188)) the async URLSession func does not invoke `URLSessionDownloadDelegate` callbacks as it only accepts a `URLSessionTaskDelegate`.

Reply:
Edit: Our implementation uses the `downloadTask(with:)` URLSession helper so it seems that the consumer could use the `URLSessionDelegate` parameter in the `APIClient.init` to pass in their `URLSessionDownloadDelegate` to receive progress callbacks.

Reply:
Correction: Our implementation actually still uses the completion handler helper of URLSession, so it seems the delegate callbacks won't be called.

Reply:
I think you misunderstood me, this protocol `AsyncDownloadRequest `, can be used as well for the non `async` version of download function, instead of having another closure for the progress, so we can just call it `DownloadRequest `